File: BraintreeModels/src/test/java/com/braintreepayments/api/KountConfigurationUnitTest.java
package com.braintreepayments.api;

import org.json.JSONException;
import org.json.JSONObject;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

import static junit.framework.Assert.assertEquals;
import static junit.framework.Assert.assertTrue;
import static org.junit.Assert.assertFalse;

@RunWith(RobolectricTestRunner.class)
public class KountConfigurationUnitTest {

    @Test
    public void fromJson_parsesFullInput() throws JSONException {
        JSONObject input = new JSONObject()
                .put("kountMerchantId", "123456");

        KountConfiguration sut = KountConfiguration.fromJson(input);
        assertTrue(sut.isEnabled());
        assertEquals("123456", sut.getKountMerchantId());
    }

    @Test
    public void fromJson_whenInputNull_returnsConfigWithDefaultValues() {
       KountConfiguration sut = KountConfiguration.fromJson(null);
       assertFalse(sut.isEnabled());
       assertEquals("", sut.getKountMerchantId());
    }

    @Test
    public void fromJson_whenInputEmpty_returnsConfigWithDefaultValues() {
        KountConfiguration sut = KountConfiguration.fromJson(new JSONObject());
        assertFalse(sut.isEnabled());
        assertEquals("", sut.getKountMerchantId());
    }
}

File: BraintreeCore/src/test/java/com/braintreepayments/api/KountConfigurationUnitTest.kt
package com.braintreepayments.api

import org.robolectric.RobolectricTestRunner
import org.json.JSONObject
import org.junit.Assert.*
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(RobolectricTestRunner::class)
class KountConfigurationUnitTest {
    
    @Test
    fun fromJson_parsesFullInput() {
        val input = JSONObject()
            .put("kountMerchantId", "123456")
        val sut = KountConfiguration(input)
        assertTrue(sut.isEnabled)
        assertEquals("123456", sut.kountMerchantId)
    }

    @Test
    fun fromJson_whenInputNull_returnsConfigWithDefaultValues() {
        val sut = KountConfiguration(null)
        assertFalse(sut.isEnabled)
        assertEquals("", sut.kountMerchantId)
    }

    @Test
    fun fromJson_whenInputEmpty_returnsConfigWithDefaultValues() {
        val sut = KountConfiguration(JSONObject())
        assertFalse(sut.isEnabled)
        assertEquals("", sut.kountMerchantId)
    }
}